velox/docs/develop/anti-join.rst

```diff
@@ -0,0 +1,581 @@
+==========
+Anti joins
+==========
+
+Anti joins can be used to efficiently implement queries with NOT IN <subquery>
+and NOT EXISTS <subquery> clauses. The semantics of these queries are different
+in the presence of NULLs in the outer query or subquery. NOT IN semantics are
+implemented by the null aware anti join. NOT EXISTS semantics are implemented
+by the regular anti join.
+
+Currently, Velox provides only null aware anti join via the JoinType::kAnti.
+Regular anti join is not available. We will rename kAnti to kNullAwareAnti for
+clarity and introduce a new kAnti join type for the regular anti join.
+
+This article explains the differences in semantics between NOT IN and NOT EXISTS
+queries and discusses the implementation of these in the null aware and regular
+anti joins.
+
+NOTE: Presto optimizer doesn’t plan NOT IN and NOT EXISTS queries using anti
+joins. To compensate, query plan translation in Prestissimo detects a NOT IN
+pattern and `converts <https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp#L1031>`_
+it into Velox anti join plan node.
```

Comment:
Links to the code will get obsolete soon.
Might be better to specify function names (or if they are the same, as we often have that in Velox, unfortunately, specify the signature.)